app/assets/javascripts/float-kit/addon/components/d-default-toast.gjs

```diff
@@ -1,63 +1,155 @@
-      (concat "-" (or @data.theme "default"))
-    }}
-    ...attributes
-  >
-    {{#if @showProgressBar}}
-      <div
-        class="fk-d-default-toast__progress-bar"
-        {{didInsert @onRegisterProgressBar}}
-      ></div>
-    {{/if}}
-    {{#if @data.icon}}
-      <div class="fk-d-default-toast__icon-container">
-        {{icon @data.icon}}
-      </div>
-    {{/if}}
-    <div class="fk-d-default-toast__main-container">
-      <div class="fk-d-default-toast__texts">
-        {{#if @data.title}}
-          <div class="fk-d-default-toast__title">
-            {{@data.title}}
-          </div>
-        {{/if}}
-        {{#if @data.message}}
-          <div class="fk-d-default-toast__message">
-            {{@data.message}}
-          </div>
-        {{/if}}
-      </div>
+const MIN_SWIPE_THRESHOLD = 10;
+
+export default class DDefaultToast extends Component {
+  @service site;
+
+  animating = false;
+  scrollLocked = false;
+  swipeEnabled = this.site.mobileView;
+
+  @action
+  setupToast(element) {
+    this.wrapperElement = element.parentElement;
+    this.wrapperElement.addEventListener(
+      "touchstart",
+      this.toggleLock.bind(this)
+    );
+  }
+
+  @action
+  handleSwipe(state) {
+    if (!this.site.mobileView || this.animating) {
+      return;
+    }
+
+    if (state.deltaY > MIN_SWIPE_THRESHOLD) {
+      this.#animateWrapperPosition();
+      return;
+    }
+  }
+
+  @action
+  handleSwipeEnded(state) {
+    if (!this.site.mobileView) {
+      return;
+    }
+
+    this.toggleLock();
+
+    if (state.deltaY > MIN_SWIPE_THRESHOLD) {
+      this.args.close();
+    }
+  }
+
+  async #animateWrapperPosition() {
+    this.animating = true;
 
-      {{#if @data.actions}}
-        <div class="fk-d-default-toast__actions">
-          {{#each @data.actions as |toastAction|}}
-            {{#if toastAction.action}}
-              <DButton
-                @icon={{toastAction.icon}}
-                @translatedLabel={{toastAction.label}}
-                @action={{fn toastAction.action (hash data=@data close=@close)}}
-                class={{toastAction.class}}
-                tabindex="0"
-              />
-            {{/if}}
-          {{/each}}
+    await this.wrapperElement.animate([{ transform: `translateY(-150px)` }], {
+      duration: 500,
+      fill: "forwards",
+    });
+
+    this.animating = false;
+  }
+
+  toggleLock() {
+    if (!this.site.mobileView) {
+      return;
+    }
+
+    if (this.scrollLocked) {
+      enableBodyScroll(this.wrapperElement);
+    } else {
+      disableBodyScroll(this.wrapperElement);
+    }
+
+    this.scrollLocked = !this.scrollLocked;
+  }
+
+  @action
+  cleanupToast() {
+    this.wrapperElement.removeEventListener(
+      "touchstart",
+      this.toggleLock.bind(this)
+    );
+  }
+
+  <template>
+    <div
+      class={{concatClass
+        "fk-d-default-toast"
+        (concat "-" (or @data.theme "default"))
+      }}
+      ...attributes
+      {{didInsert this.setupToast}}
```

Comment:
for cleanupToast I suggest you make setupToast a modifier so you can have didInsert/willDestroy in one function:


```
setupToast = modifier(... => {
  addListener(...)
  
  return () => {
    removeLsitener(...);
  };
});

Reply:
I forgot to add the willDestroy for the cleanup, but I can use the modifier approach to handle this instead 👍